client/client.go

```diff
@@ -137,3 +145,19 @@ func FromContext(ctx context.Context) Info {
 	}
 	return c
 }
+
+// NewMetadata creates a new Metadata object to use in Info. md is used as-is.
+func NewMetadata(md map[string][]string) Metadata {
+	return Metadata{
+		data: md,
+	}
+}
+
+// Get gets the value of the key from metadata, returning a copy.
+func (m Metadata) Get(key string) []string {
+	vals := m.data[key]
+	ret := make([]string, 0, len(vals))
+	copy(ret, vals)
+
+	return ret
```

Comment:
```suggestion
	vals, found := m.data[key]
	
	if !found {
		return []string{}
	}
	
	ret := make([]string, len(vals))
	copy(ret, vals)

	return ret
```
The main issue is that copy is limited by len(dest) not cap(dest), an the current implementation will always return an empty slice.

Reply:
Thanks @trasc! fixed